### PR TITLE
fix: Named Accessors, edge case with Optional types

### PR DESCRIPTION
**Pull request by HarryET**
Edge case involving optional types, looks like types are not checked correctly

### CODE REVIEW

File: compiler-core/src/javascript/tests/records.rs [hunk 1]
```diff
@@ -53,3 +53,18 @@ pub type Person {
...
+    assert_js!(
+        "
+pub type Person {
+    Teacher(name: String, age: Option(Int), title: String)
+    Student(name: String, age: Option(Int))
+}
+pub fn get_name(person: Person) { person.name }
+pub fn get_age(person: Person) { person.age }"
+    );
+}
```
Comment (lpil) [inline compiler-core/src/javascript/tests/records.rs:70]: End of file newline please.

In vscode this is `"files.insertFinalNewline": true`, good to have it globally as text files on *nix system should always have them.

Comment (HarryET) [reply compiler-core/src/javascript/tests/records.rs]: Fixed

**Review by lpil — APPROVED** [review]
Thank Harry!
